### ISSUE TITLE
Findbugs detect implementing raw type of generic interface.

### ISSUE THREAD

**Issue by mehwhatever** [spotbugs/spotbugs#1053]
Currently FindBugs doesn't raise a bug if a class implements a raw type of a generic interface.

Is there an annotation which can be put on an interface to allow FindBugs to raise an issue if any of it's implementation implements it raw type.

For eg for this interface,

```
@DoNotAllowRawImplemetation
class interface Interface<T extends SomeClass>
```

```
class Impl implements Interface
// this should raise a bug.
```

Wanted to know if something like this exists and if not thoughts on if this should be a part of Findbugs?

**Comment by ThrawnCA** [comment]
Isn't that already a compiler warning?

**Comment by mehwhatever** [comment]
@ThrawnCA Yes but I would like to break builds on this.

**Comment by ThrawnCA** [comment]
Well, there's the compiler `-Werror` flag.

**Comment by mehwhatever** [comment]
That is a 1 way door and very inflexible way. I can either have all warnings as errors or none of them.

Also my suggestion was that people can mark *some* interfaces which they don't want the implementations to be of ray type.

**Comment by mehwhatever** [comment]
Any thoughts on this?

**Comment by mehwhatever** [comment]
Any update? Not sure if this actively monitored.
